src/app/main/component/user/components/habit/add-new-habit/habit-edit-shopping-list/edit-shopping-list.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { ShoppingList } from '@global-user/models/shoppinglist.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EditShoppingListService {
+  public list = [];
+  public list$ = new Subject();
+
+  public fillList(data: ShoppingList[]) {
+    this.list = data;
+    this.list$.next(this.list);
+  }
+
+  public getList(): Observable<any> {
+    return this.list$.asObservable();
```

Comment:
why do you need to do this ?

Reply:
I cant found better solution for loading shopping list asynchronously. And one more reson that we will have ability to add custom items in shopping list and also delete it.